lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1087 @@
+  case SimpleTypeKind::Int128:
+    return model::PrimitiveTypeKind::Signed;
+  case SimpleTypeKind::Float16:
+  case SimpleTypeKind::Float32:
+  case SimpleTypeKind::Float64:
+  case SimpleTypeKind::Float80:
+  case SimpleTypeKind::Float128:
+    return model::PrimitiveTypeKind::Float;
+  default:
+    return model::PrimitiveTypeKind::Invalid;
+  }
+}
+
+void PDBImporterTypeVisitor::createPrimitiveType(TypeIndex TI) {
+  using namespace model;
+  PrimitiveTypeKind::Values Kind = codeviewSimpleTypeEncodingToModel(TI);
+
+  if (TI.getSimpleKind() == SimpleTypeKind::Void) {
+    auto VoidType = Model->getPrimitiveType(PrimitiveTypeKind::Void, 0);
+
+    if (TI.getSimpleMode() == SimpleTypeMode::Direct)
+      ProcessedTypes[TI] = VoidType;
+    else {
+      // Pointer.
+      auto TypedefTy = makeType<TypedefType>();
+      auto TypeDef = cast<TypedefType>(TypedefTy.get());
+      std::vector<Qualifier> Qualifiers;
+      if (TI.getSimpleMode() == SimpleTypeMode::NearPointer32) {
+        constexpr uint64_t PointerSize = 4;
+        Qualifiers.push_back({ Qualifier::createPointer(PointerSize) });
+      } else if (TI.getSimpleMode() == SimpleTypeMode::NearPointer64) {
+        constexpr uint64_t PointerSize = 8;
+        Qualifiers.push_back({ Qualifier::createPointer(PointerSize) });
+      }
+
+      QualifiedType Qty(VoidType, Qualifiers);
+      TypeDef->UnderlyingType = Qty;
+      auto TyPath = Model->recordNewType(std::move(TypedefTy));
+      ProcessedTypes[TI] = TyPath;
+    }
+    return;
+  }
+
+  auto TySize = getSizeinBytes(TI);
+  // TODO: Create an InvalidTypePlaceHolder as DwarfImporter.
+  if (TySize == 0 or Kind == PrimitiveTypeKind::Invalid) {
+    return;
+  }
+
+  // Remember the type.
+  auto PrimitiveModelType = Model->getPrimitiveType(Kind, TySize);
+  ProcessedTypes[TI] = PrimitiveModelType;
+}
+
+// ==== Implementation of the Model Symbol-type connection ==== //
+
+Error PDBImporterSymbolVisitor::visitSymbolBegin(CVSymbol &Record) {
+  return visitSymbolBegin(Record, 0);
+}
+
+Error PDBImporterSymbolVisitor::visitSymbolBegin(CVSymbol &Record,
+                                                 uint32_t Offset) {
+  return Error::success();
+}
+
+Error PDBImporterSymbolVisitor::visitKnownRecord(CVSymbol &Record,
+                                                 ProcSym &Proc) {
+  // If it is not in the .idata already, we assume it is a static symbol.
+  if (not Model->ImportedDynamicFunctions.count(Proc.Name.str())) {
+    uint64_t FnVA = Session.getRVAFromSectOffset(Proc.Segment, Proc.CodeOffset);
+    // Relocate the symbol.
+    MetaAddress FnAddress = ImageBase + FnVA;
+
+    if (not Model->Functions.count(FnAddress)) {
+      model::Function &Function = Model->Functions[FnAddress];
+      Function.OriginalName = Proc.Name;
+      TypeIndex TyFn = Proc.FunctionType;
+      if (ProcessedTypes.count(TyFn)) {
+        model::QualifiedType Qty(ProcessedTypes[TyFn], {});
+        Function.Prototype = Qty.UnqualifiedType;
```

Comment:
Any reason not do assign it inline?

Reply:
When the code is improved, the `revng-check-conventions` will force us to be using local variables here.